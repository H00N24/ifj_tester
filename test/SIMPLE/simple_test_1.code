class Main
{
	static int iter = 5;
	static void run()
	{
		while(iter > 0)
			iterator.processValue();

		iterator.printStatus(iter);
		iter = 6;

		while(iter > 1)
			iterator.processValue();

		iterator.printStatus(iter);
		iter = 6;

		while(iter > 2)
			iterator.processValue();

		iterator.printStatus(iter);
		iter = 10;

		while(iter > 4)
		{
			iterator.processValue();
			iterator.processValue();
		}

		iterator.printStatus(iter);
	}
}

class iterator
{
	static void processValue()
	{
		ifj16.print(Main.iter + "\n");
		Main.iter = Main.iter - 1;
	}
	static void printStatus(int iter)
	{
		if(iter == 0)
			ifj16.print("Success!\n");
		else if(iter == 1)
				ifj16.print("Fail 1\n");
		else if(iter == 2)
				ifj16.print("Fail 2\n");
		else if(iter > 2)
		{
			ifj16.print("Fail > 2\n");
			return;
		}
	}
}
