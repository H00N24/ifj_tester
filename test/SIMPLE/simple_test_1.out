5
4
3
2
1
Success!
6
5
4
3
2
Fail 1
6
5
4
3
Fail 2
10
9
8
7
6
5
Fail > 2
